### ISSUE TITLE
Incorrect LambdaMethodReference error-prone warning

### ISSUE THREAD

**Issue by dtobin** [palantir/gradle-baseline#772]
## What happened?

Lambda expression:
```
() -> foo.getBar().baz()
```
Warning printed:
```
Note: [LambdaMethodReference] Lambda should be a method reference
```

version: gradle-baseline-java 1.16.0

## What did you want to happen?

No warning printed. Unless I'm missing something, I don't see a way to convert this expression to a method reference. This would be functionally not equivalent:
```
foo.getBar()::baz
```

**Comment by carterkozak** [comment]
Have you seen https://github.com/palantir/gradle-baseline/pull/771

**Comment by dtobin** [comment]
Ah! Requiring static methods will fix this issue, though is there any reason we would want to rewrite (or suggest a rewrite for) chained method calls in a lambda at all?

**Comment by carterkozak** [comment]
Nope, chained method invocations cannot result in a static method invocation.

**Comment by dtobin** [comment]
I guess a better phrasing of my question is why were chained method invocations considered in the first place (before the switch to only allowing static method invocations), and should we add an additional check to the code to immediately return on chained method invocations, in case you eventually revert the static method restriction? In any case, this issue will be fixed by #771, no additional work required.
